forms-flow-web/src/components/Modals/TaskVariableModal.js

```diff
@@ -356,58 +362,104 @@ const TaskVariableModal = React.memo(
+                <CustomInfo
+                  heading="Note"
+                  content="To use variables in the flow, as well as sorting by them in 
+                  the submissions and tasks you need to specify which variables you want to import from the layout. Variables get imported into the system at the time of the submission, if the variables that are needed 
+                 are not selected prior to the form submission THEY WILL NOT BE AVAILABLE in the flow, submissions, and tasks."
+                />
+                <div>
+                  <label className="selected-var-text">
+                    {t("Selected Variables")}
+                  </label>
+                  <PillList
+                    alternativeLabels={alternativeLabels}
+                    onRemove={removeSelectedVariable}
+                  />
+                </div>
+              </div>
+              <div className="variable-container">
+                <FormComponent
+                  form={form}
+                  alternativeLabels={alternativeLabels}
+                  setAlternativeLabels={setAlternativeLabels}
+                  setSelectedComponent={setSelectedComponent}
+                  selectedComponent={selectedComponent}
+                />
+              </div>
+            </>
+          )}
         </Modal.Body>
         <Modal.Footer>
-          <CustomButton
-            variant="primary"
-            size="md"
-            className=""
-            disabled={isPublished}
-            label={t("Save")}
-            ariaLabel="save task variable btn"
-            dataTestid="save-task-variable-btn"
-            onClick={handleSaveTaskVariable}
-          />
-          <CustomButton
-            variant="secondary"
-            size="md"
-            className=""
-            label={t("Cancel")}
-            ariaLabel="Cancel btn"
-            dataTestid="Cancel-btn"
-            onClick={handleClose}
-          />
+        {layoutNotsaved ? (
+    // Footer content when layoutNotsaved is true
+    <>
+    <CustomButton
+      variant="primary"
+      size="md"
+      className=""
+      label={t("Back to Layout")}
+      ariaLabel="Back to Layout btn"
+      dataTestid="back-to-layout-btn"
+      onClick={handleClose}
+    />
+    <CustomButton
+        variant="secondary"
+        size="md"
+        className=""
+        label={t("Cancel")}
+        ariaLabel="Cancel btn"
+        dataTestid="cancel-btn"
+        onClick={handleClose}
+      />
+    </>
+  ) : (
+    // Footer content when layoutNotsaved is false
+    <>
+      <CustomButton
+        variant="primary"
+        size="md"
+        className=""
+        disabled={isPublished}
+        label={t("Save")}
+        ariaLabel="save task variable btn"
+        dataTestid="save-task-variable-btn"
+        onClick={handleSaveTaskVariable}
+      />
+      <CustomButton
+        variant="secondary"
+        size="md"
+        className=""
```

Comment:
Could you please separate these two pieces of content into two variables and use a ternary operator to conditionally render them, instead of placing everything together inside the return statement?